packages/aws-imds/src/index.ts

```diff
@@ -0,0 +1,75 @@
+import fetch from 'node-fetch';
+import { z } from 'zod';
+
+const IMDS_URI = 'http://169.254.169.254';
+const TOKEN_PATH = '/latest/api/token';
+const TOKEN_TTL = 21_600; // 6 hours
+
+const InstanceIdentitySchema = z.object({
+  availabilityZone: z.string(),
+  privateIp: z.string(),
+  version: z.string(),
+  instanceId: z.string(),
+  instanceType: z.string(),
+  accountId: z.string(),
+  imageId: z.string(),
+  pendingTime: z.string(),
+  architecture: z.string(),
+  region: z.string(),
+});
```

Comment:
Just to confirm my understanding: if AWS adds more properties to the instance metadata in the future we'll still correctly parse it, because those extra properties will only be narrowing the type, correct? That is, we won't suddenly break because AWS adds a property?

Reply:
Correct! We'll just silently drop them and they'll be inaccessible to anything downstream of this package.

Reply:
Perfect!